nextcord/gateway.py

```diff
@@ -303,52 +304,54 @@ class DiscordWebSocket:
         shard_count: int | None
         _max_heartbeat_timeout: float
 
-    def __init__(self, socket, *, loop):
-        self.socket = socket
-        self.loop = loop
+    def __init__(
+        self, socket: aiohttp.ClientWebSocketResponse, *, loop: asyncio.AbstractEventLoop
+    ) -> None:
+        self.socket: aiohttp.ClientWebSocketResponse = socket
+        self.loop: asyncio.AbstractEventLoop = loop
 
         # an empty dispatcher to prevent crashes
         self._dispatch: VariadicArgNone = lambda *args: None
         # generic event listeners
-        self._dispatch_listeners = []
+        self._dispatch_listeners: List[EventListener] = []
         # the keep alive
-        self._keep_alive = None
-        self.thread_id = threading.get_ident()
+        self._keep_alive: Optional[KeepAliveHandler] = None
+        self.thread_id: int = threading.get_ident()
 
         # ws related stuff
-        self.session_id = None
-        self.resume_url = None
-        self.sequence = None
-        self._zlib = zlib.decompressobj()
-        self._buffer = bytearray()
-        self._close_code = None
-        self._rate_limiter = GatewayRatelimiter()
+        self.session_id: Optional[str] = None
+        self.resume_url: Optional[str] = None
+        self.sequence: Optional[int] = None
+        self._zlib: zlib._Decompress = zlib.decompressobj()
```

Comment:
Pyright Strict will give errors about accessing a private attribute for this type hint. It's fine to remove it.